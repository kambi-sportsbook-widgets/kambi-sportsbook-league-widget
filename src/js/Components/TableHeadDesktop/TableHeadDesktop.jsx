import React from 'react'
import PropTypes from 'prop-types'
import TableHeadDefaultRow from '../TableHeadDefaultRow/TableHeadDefaultRow'

const TableHeadDesktop = ({ columns, title }) => (
  <thead>
    <TableHeadDefaultRow columns={columns} title={title} />
  </thead>
)

TableHeadDesktop.propTypes = {
  /**
   * Widget's title
   */
  title: PropTypes.string.isRequired,

  /**
   * Definitions of column groups
   */
  columns: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      short: PropTypes.string.isRequired,
    })
  ).isRequired,
}

export default TableHeadDesktop
