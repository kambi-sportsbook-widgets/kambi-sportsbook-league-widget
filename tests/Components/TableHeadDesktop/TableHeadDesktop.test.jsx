/* eslint-env jest */
import React, { Children } from 'react';
import TableHeadDesktop from '../../../src/js/Components/TableHeadDesktop/TableHeadDesktop';
import ReactTestRenderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('TableHeadDesktop DOM rendering', () => {

   it('renders correctly when not collapsable', () => {
      const tree = ReactTestRenderer.create(
         <TableHeadDesktop
            title="Test title"
            columns={[{name: 'Test column 1', short: 'TC1'}]}
            onHeadClick={() => false}
            collapsable={false}
         />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly when collapsable', () => {
      const tree = ReactTestRenderer.create(
         <TableHeadDesktop
            title="Test title"
            columns={[{name: 'Test column 1', short: 'TC1'}]}
            onHeadClick={() => false}
            collapsable={true}
         />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

   it('renders correctly when in hidden mode', () => {
      const tree = ReactTestRenderer.create(
         <TableHeadDesktop
            title="Test title"
            columns={[{name: 'Test column 1', short: 'TC1'}]}
            onHeadClick={() => false}
            collapsable={true}
            hiddenMode={true}
         />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

});

describe('TableHeadDesktop interface', () => {

   it('handles head click events correctly', () => {
      const onHeadClickMock = jest.fn();

      const wrapper = shallow(
         <TableHeadDesktop
            title="Test title"
            columns={[{name: 'Test column 1', short: 'TC1'}]}
            onHeadClick={onHeadClickMock}
            collapsable={true}
         />
      );

      expect(onHeadClickMock).not.toHaveBeenCalled();

      wrapper.first().simulate('click');

      expect(onHeadClickMock).toHaveBeenCalledTimes(1);
   });

});
