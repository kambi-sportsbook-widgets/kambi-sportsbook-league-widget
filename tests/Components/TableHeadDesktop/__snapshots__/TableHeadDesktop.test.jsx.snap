// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeadDesktop DOM rendering renders correctly when collapsable 1`] = `
<thead
  onClick={[Function]}
>
  <tr
    className="KambiWidget-header general"
  >
    <th
      className="title"
      colSpan="2"
    >
      Test title
    </th>
    <th
      className="column-name"
      title="Test column 1"
    >
      TC1
    </th>
    <th
      className="margin"
    />
  </tr>
</thead>
`;

exports[`TableHeadDesktop DOM rendering renders correctly when in hidden mode 1`] = `
<thead
  onClick={[Function]}
>
  <tr
    className="KambiWidget-header general"
  >
    <th
      className="title"
      colSpan="2"
    >
      Test title
    </th>
    <th
      className="column-name"
      title="Test column 1"
    >
      
    </th>
    <th
      className="margin"
    />
  </tr>
</thead>
`;

exports[`TableHeadDesktop DOM rendering renders correctly when not collapsable 1`] = `
<thead
  onClick={[Function]}
>
  <tr
    className="KambiWidget-card-header-border KambiWidget-secondary-header KambiWidget-card-support-text-color general general-non-collapsable"
  >
    <th
      className="title"
      colSpan="2"
    >
      Test title
    </th>
    <th
      className="column-name"
      title="Test column 1"
    >
      TC1
    </th>
    <th
      className="margin"
    />
  </tr>
</thead>
`;
